include/mata/nfa.hh

```diff
@@ -196,22 +221,39 @@ struct Post : private Util::OrdVector<Move> {
 
     virtual ~Post() = default;
 
-    const_iterator find(const Move& m) const override { return Util::OrdVector<Move>::find(m);}
-    iterator find(const Move& m) override { return Util::OrdVector<Move>::find(m);}
+    //Trying to make Past movable, oes it make sense?
+    Post(Post&& rhs) = default;
+    Post(const Post& rhs) = default;
+    Post & operator=(Post&& rhs) = default;
+    Post & operator=(const Post& rhs) = default;
+
+    inline const_iterator find(const Move& m) const override { return Util::OrdVector<Move>::find(m);}
+    inline iterator find(const Move& m) override { return Util::OrdVector<Move>::find(m);}
+
+    void inline insert(const Move& m) override { Util::OrdVector<Move>::insert(m); }
+
+    // dangerous, breaks the sortedness invariant
+    void inline push_back(const Move& m) override { Util::OrdVector<Move>::push_back(m); }
+
+    inline const Move& back() const override { return Util::OrdVector<Move>::back(); }
 
-    void insert(const Move& m) override { Util::OrdVector<Move>::insert(m); }
+    // is adding this non-const version ok?
+    inline Move& back() { return Util::OrdVector<Move>::back(); }
```

Comment:
This might be dangerous, you can now change stuff at the end, potentionally breaking the sortedness.

Reply:
This would require similar handling as for `push_back`. However, we should probably have a `mutable_back()` function to explicitly state that you get a mutable reference. That would be consistent with the `get_mutable_post` method.

Reply:
Can you really change things using that? Maybe make it const?

Reply:
anybody wants to do something about this?